packages/core/src/internal/execution/nonce-management/get-nonce-sync-messages.ts

```diff
@@ -98,9 +101,14 @@ export async function getNonceSyncMessages(
       "latest"
     );
 
+    const hasPendingTransactions =
+      safeConfirmationsCount === undefined
+        ? pendingCount > 0
+        : safeConfirmationsCount !== pendingCount;
+
     // Case 0: We don't have any pending transactions
     if (pendingTransactions.length === 0) {
-      if (safeConfirmationsCount !== pendingCount) {
+      if (hasPendingTransactions) {
```

Comment:
We need to differentiate the names of the variables, so it is clear that we mean `ignition tracked pending transactions` and `onchainPendingCount`

Reply:
maybe `hasPendingTransactions` should become `hasPendingTransactionsThatHaveNotConfirmed` or some such?